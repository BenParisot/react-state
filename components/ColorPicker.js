import React, { PureComponent } from 'react';
import ColorDisplay from './ColorDisplay';

export default class ColorPicker extends PureComponent {
 state = {
   selectedColor: 'black'
 }

 clickHandler = event => {
   this.setState({ selectedColor: event.target.textContent }); console.log(event.target.textContent);
 }

 render() {
   return (
            <>
            <button onClick={this.clickHandler} style={{ background:'red' }}>red</button>
            <button onClick={this.clickHandler} style={{ background:'blue' }}>blue</button>
            <button onClick={this.clickHandler} style={{ background:'yellow' }}>yellow</button>
            <ColorDisplay color={this.state.selectedColor} />
          </>
   );
 }
}
