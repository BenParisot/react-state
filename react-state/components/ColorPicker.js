import React, { PureComponent } from 'react';

export default class ColorPicker extends PureComponent {
    state = {
      background: 'grey',
      height: '100px',
      width: '100px'
    }

    colors = ['red', 'blue', 'yellow']

    clickHandler = event => {
      this.setState({ background: event.target.textContent });
    }
    
    render() {
      const buttons = this.colors.map(color => {
        return <button key={color} onClick={this.clickHandler} style={{ background: color }}>{color}</button>;

      });

      return (
        <>
            {buttons}
            <div style={this.state}></div>
        </>
      );
    }
}
