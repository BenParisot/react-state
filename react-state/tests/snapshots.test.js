import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from '../components/ColorPicker';

describe('ColorPicker component', () => {
  it('pickes a Color', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
