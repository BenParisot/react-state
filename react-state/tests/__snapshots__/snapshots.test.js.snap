// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component counts 1`] = `
<Fragment>
  <p>
    Total Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Click me to count!
  </button>
</Fragment>
`;
